src/Adapter/MSTest.CoreAdapter/Execution/TestClassInfo.cs

```diff
@@ -306,40 +311,52 @@ public string RunClassCleanup()
                 return null;
             }
 
-            lock (this.testClassExecuteSyncObject)
+            if (this.IsClassInitializeExecuted)
             {
-                try
-                {
-                    this.ClassCleanupMethod.InvokeAsSynchronousTask(null);
-
-                    return null;
-                }
-                catch (Exception exception)
+                lock (this.testClassExecuteSyncObject)
```

Comment:
lock can be removed? All clean ups done in [single thread](https://github.com/Microsoft/testfx/blob/0789f7a8c1854f406e6ce93dd0bf468a1ec5ef52/src/Adapter/MSTest.CoreAdapter/Execution/TestExecutionManager.cs#L337). /cc @AbhitejJohn